osu.Game/Skinning/SkinReloadableDrawable.cs

```diff
@@ -56,6 +68,22 @@ protected virtual void SkinChanged(ISkinSource skin)
         {
         }
 
+        private void onChange()
+        {
+            // schedule required to avoid calls after disposed.
+            // note that this has the side-effect of components only performing a skin change when they are alive.
+            pendingSkinChange?.Cancel();
+            pendingSkinChange = Scheduler.Add(skinChanged);
```

Comment:
Should I care that this changed from `Scheduler.AddOnce()` to `Scheduler.Add()`? It seems potentially important (and potentially performance-regressive).

Reply:
I tend to agree, I think `??=` should fix that?

Reply:
On second thought I'm not sure what my point here was, since even if the event fires multiple times between frames, the intermediate delegates will be canceled anyway.

Probably fine to keep this as is. (`??=` does nothing here anyways.)

Reply:
I still prefer doing `pendingSkinChange ??= Scheduler.Add(skinChanged)`, because as you said, it is a parameter-less function, so it doesn't really matter which skin change scheduled it, as long as there is something scheduled.

Reply:
As written `??=` would be plain wrong.

1. Create a delegate
2. Cancel it once
3. If it's not nulled by something (which it won't be if it's canceled before it's ran), then nothing will get scheduled ever again

Reply:
If we can make `AddOnce` work again, that would be optimal. But I believe the reason I switched away from it was due to it nor returning the delegate. Will require some further investigation.